moto/dynamodb/models/table.py

```diff
@@ -465,7 +465,7 @@ def hash_key_names(self) -> List[str]:
 
     @property
     def range_key_names(self) -> List[str]:
-        keys = [self.range_key_attr]
+        keys = [self.range_key_attr] if self.has_range_key else []
```

Comment:
Unrelated cleanup. In practice, we always gate this behind `if self.has_range_key:`. 
But if we call this property without verifying that a range key exists, it would return `[None]`. 
Returning an empty key-list makes more sense.